src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java

```diff
@@ -0,0 +1,1202 @@
+            setAchTransactions(achIncomeFile, achIncomeTransactions);
+        }
+
+        saveTransaction(achIncomeTransactions);
+    }
+
+    private void saveTransaction(List achIncomeFileTransactions) {
+        businessObjectService.save(achIncomeFileTransactions);
+    }
+
+    /**
+     * Calls businessObjectService to remove all the ach Income transaction rows from the transaction load table.
+     */
+    public void cleanTransactionsTable() {
+        businessObjectService.deleteMatching(AchIncomeTransaction.class, new HashMap());
+    }
+
+    protected void setAchTransactions(AchIncomeFile achIncomeFile, List achIncomeTransactions) {
+        Timestamp bankTimestamp = getFormattedTimestamp(achIncomeFile, "fileDate/time");
+
+        for (AchIncomeFileGroup achIncomeFileGroup : achIncomeFile.getGroups()) {
+            for (AchIncomeFileTransactionSet achIncomeFileTransactionSet : achIncomeFileGroup.getTransactionSets()) {
+                for (AchIncomeFileTransaction achIncomeFileTransaction : achIncomeFileTransactionSet.getTransactionGuts()) {
+                    if (KFSConstants.GL_CREDIT_CODE.equals(achIncomeFileTransaction.getCreditDebitIndicator())) {
+                        AchIncomeTransaction achIncomeTransaction = new AchIncomeTransaction();
+                        achIncomeTransaction.setPaymentMethodCode(achIncomeFileTransaction.getPaymentMethodCode());
+                        achIncomeTransaction.setEffectiveDate(achIncomeFileTransaction.getEffectiveDate());
+                        achIncomeTransaction.setLoadTimestamp(dateTimeService.getCurrentTimestamp());
+                        achIncomeTransaction.setBankTimestamp(bankTimestamp);
+                        achIncomeTransaction.setTransactionAmount(achIncomeFileTransaction.getTransactionAmount());
+                        achIncomeTransaction.setPayerName(truncatePayerNameIfNecessary(achIncomeFileTransaction));
+                        achIncomeTransaction.setNotes(createNotes(achIncomeFileTransaction));
+                        achIncomeTransactions.add(achIncomeTransaction);
+                    }
+                }
+            }
+        }
+    }
+
+    protected String truncatePayerNameIfNecessary(AchIncomeFileTransaction achIncomeFileTransaction) {
+        String payerName = achIncomeFileTransaction.getPayerName();
+
+        if (payerName.length() > CuFPConstants.AchIncomeFileTransactionPayerOrPayeeName.ACH_TRN_PAYR_NM_DB_SIZE) {
+            payerName = payerName.substring(0, CuFPConstants.AchIncomeFileTransactionPayerOrPayeeName.ACH_TRN_PAYR_NM_DB_SIZE);
+        }
+
+        return payerName;
+    }
+
+    private List<AchIncomeNote> createNotes(AchIncomeFileTransaction achIncomeFileTransaction) {
+        StringBuffer notes = addPaymentInformationToNotes(achIncomeFileTransaction);
+        List<String> achNotes = addRMRAndNTELinesToNotes(achIncomeFileTransaction, notes);
+
+        List<AchIncomeNote> achIncomeNotes = new ArrayList<>();
```

Comment:
yes, that's the preferred way for Java 7 and above; it leverages what is defined with the type